py-polars/polars/dataframe/frame.py

```diff
@@ -3618,44 +3639,82 @@ def write_delta(
         ...     },
         ... )  # doctest: +SKIP
 
+        Merging dataframe to the local filesystem as a Delta Lake table. For a
+        cloud object store just pass storage_options or a DeltaTable object.
+        For all additional TableMerger methods check the deltalake docs `here <https://delta-io.github.io/delta-rs/python/api_reference.html#deltalake.table.TableMerger`__.
+
+        Schema evolution is currently not yet support in deltalake, therefore
+        overwrite_schema won't have any effect during `MERGE`.
+
+        >>> df = pl.DataFrame(
+        ...     {
+        ...         "foo": [1, 2, 3, 4, 5],
+        ...         "bar": [6, 7, 8, 9, 10],
+        ...         "ham": ["a", "b", "c", "d", "e"],
+        ...     }
+        ... )
+        >>> table_path = "/path/to/delta-table/"
+        >>> (
+        ...     df.write_delta(
+        ...         "table_path",
+        ...         mode="merge",
+        ...         delta_merge_options={
+        ...             "predicate": "s.foo = t.foo",
+        ...             "source_alias": "s",
+        ...             "target_alias": "t",
+        ...         },
+        ...     )
+        ...     .when_matched_update_all()
+        ...     .when_not_matched_insert_all()
+        ...     .execute()
+        ... )  # doctest: +SKIP
         """
         from polars.io.delta import (
             _check_for_unsupported_types,
             _check_if_delta_available,
-            _convert_pa_schema_to_delta,
             _resolve_delta_lake_uri,
         )
 
         _check_if_delta_available()
 
-        from deltalake.writer import write_deltalake
+        from deltalake import DeltaTable, write_deltalake
 
-        if delta_write_options is None:
-            delta_write_options = {}
+        _check_for_unsupported_types(self.dtypes)
 
         if isinstance(target, (str, Path)):
             target = _resolve_delta_lake_uri(str(target), strict=False)
 
-        _check_for_unsupported_types(self.dtypes)
-
         data = self.to_arrow()
 
-        schema = delta_write_options.pop("schema", None)
-        if schema is None:
-            schema = _convert_pa_schema_to_delta(data.schema)
+        if mode == "merge":
+            if delta_merge_options is None:
+                raise ValueError(
+                    "You need to pass delta_merge_options with at least a given predicate for `MERGE` to work."
+                )
```

Comment:
Do we need this check here or does DeltaTable.merge already throw this error?

Reply:
It does when you provide an empty dictionary, `TypeError: DeltaTable.merge() missing 1 required positional argument: 'predicate'`, but when you don't pass any value to delta_merge_options, you get this error which `TypeError: deltalake.table.DeltaTable.merge() argument after ** must be a mapping, not NoneType`, which does not clarify exactly the issue for the user